### PR TITLE
feat: Show alert about revoked certificate WPB-7211

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-7211" title="WPB-7211" target="_blank">TaskWPB-7211</a>  [iOS] Alert about revoked certificate
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

When the user discovers that the current device's certificate has been revoked, we should notify this with an alert:
image-20240320-112443

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-sync-engine/Source/E2EI/CRL/CertificateRevocationListsChecker.swift [hunk 1]
```diff
@@ -112,6 +109,7 @@ public class CertificateRevocationListsChecker: CertificateRevocationListsChecki
 
     private func checkCertificateRevocationLists(from distributionPoints: Set<URL>) async {
 
+        var checkE2EICertificateForSelfClient = false
```
Comment (netbe) [inline wire-ios-sync-engine/Source/E2EI/CRL/CertificateRevocationListsChecker.swift]: maybe this name makes more sense
```suggestion
        var shouldNotifyAboutRevokedCertificate = false
```

**Review by netbe — APPROVED** [review]
just a a suggestion

File: wire-ios-sync-engine/Source/E2EI/CRL/CertificateRevocationListsChecker.swift [hunk 2]
```diff
@@ -112,6 +109,7 @@ public class CertificateRevocationListsChecker: CertificateRevocationListsChecki
 
     private func checkCertificateRevocationLists(from distributionPoints: Set<URL>) async {
 
+        var shouldNotifyAboutRevokedCertificate = false
```
Comment (David-Henner) [inline wire-ios-sync-engine/Source/E2EI/CRL/CertificateRevocationListsChecker.swift:112]: did you mean to rename this `checkE2EICertificateForSelfClient` ?

Comment (David-Henner) [reply wire-ios-sync-engine/Source/E2EI/CRL/CertificateRevocationListsChecker.swift:112]: Ah no I see it's the other way around... You applied @netbe's suggestion but it didn't apply to the other occurrences of `checkE2EICertificateForSelfClient` so it won't be able to build

**Review by David-Henner — APPROVED** [review]
Approving now so you can merge as soon as the issue above is fixed

**Pull request merged**